sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/OpenTelemetry.Instrumentation.Http/HttpClient_TracerProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,120 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Instrumentation.Http;
+using OpenTelemetry.Instrumentation.Http.Implementation;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Extension methods to simplify registering of HttpClient instrumentation.
+    /// </summary>
+    internal static class HttpClient_TracerProviderBuilderExtensions
+    {
+        /// <summary>
+        /// Enables HttpClient instrumentation.
+        /// </summary>
+        /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+        /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+        public static TracerProviderBuilder AddHttpClientInstrumentation(this TracerProviderBuilder builder)
+            => AddHttpClientInstrumentation(builder, name: null, configureHttpClientInstrumentationOptions: null);
+
+        /// <summary>
+        /// Enables HttpClient instrumentation.
+        /// </summary>
+        /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+        /// <param name="configureHttpClientInstrumentationOptions">Callback action for configuring <see cref="HttpClientInstrumentationOptions"/>.</param>
+        /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+        internal static TracerProviderBuilder AddHttpClientInstrumentation(
+            this TracerProviderBuilder builder,
+            Action<HttpClientInstrumentationOptions> configureHttpClientInstrumentationOptions)
+            => AddHttpClientInstrumentation(builder, name: null, configureHttpClientInstrumentationOptions);
+
+        /// <summary>
+        /// Enables HttpClient instrumentation.
+        /// </summary>
+        /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+        /// <param name="name">Name which is used when retrieving options.</param>
+        /// <param name="configureHttpClientInstrumentationOptions">Callback action for configuring <see cref="HttpClientInstrumentationOptions"/>.</param>
+        /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+        internal static TracerProviderBuilder AddHttpClientInstrumentation(
+            this TracerProviderBuilder builder,
+            string name,
+            Action<HttpClientInstrumentationOptions> configureHttpClientInstrumentationOptions)
+        {
+            Guard.ThrowIfNull(builder);
+
+            // Note: Warm-up the status code mapping.
+            _ = TelemetryHelper.BoxedStatusCodes;
```

Comment:
do we need to check if user has brought up beta dependency and configured custom HTTP instrumentation options and back off ?

(same question for other instrumentations)

It's totally fine if you want to do it in a separate PR.